server/paymentProviders/stripe/creditcard.js

```diff
@@ -486,13 +486,155 @@ async function closeDispute(event) {
+    // eslint-disable-next-line camelcase
+    where: { data: { charge: { payment_intent: stripePaymentIntentId } } },
+    include: [
+      {
+        model: models.Order,
+        required: true,
+        include: [models.Subscription],
+      },
+    ],
+  });
+
+  if (!paymentIntentTransaction) {
+    return;
+  }
+
+  const transactions = await models.Transaction.findAll({
+    where: {
+      TransactionGroup: paymentIntentTransaction.TransactionGroup,
+    },
+    include: {
+      model: models.Order,
+      required: true,
+      include: [models.Subscription],
+    },
+  });
+
+  await Promise.all(
+    transactions.map(async transaction => {
+      await transaction.update({ isInReview: true });
+      const order = transaction.Order;
+      if (order.status !== OrderStatuses.IN_REVIEW) {
+        await order.update({ status: OrderStatuses.IN_REVIEW });
+
+        if (order.SubscriptionId) {
+          await order.Subscription.deactivate();
+        }
+      }
+    }),
+  );
+}
+
+// Charge on Stripe had a fraud review closed (either approved/refunded)
+async function closeReview(event) {
+  const stripePaymentIntentId = event.data.object.payment_intent;
+  const closedReason = event.data.object.closed_reason;
+
+  const paymentIntentTransaction = await models.Transaction.findOne({
+    // eslint-disable-next-line camelcase
+    where: { data: { charge: { payment_intent: stripePaymentIntentId } } },
+    include: [
+      {
+        model: models.Order,
+        required: true,
+        include: [models.Subscription],
+      },
+      {
+        model: models.User,
+        required: true,
+        as: 'createdByUser',
+      },
+    ],
+  });
+
+  if (!paymentIntentTransaction) {
+    return;
+  }
+
+  const user = paymentIntentTransaction.createdByUser;
+
+  const transactions = await models.Transaction.findAll({
+    where: {
+      TransactionGroup: paymentIntentTransaction.TransactionGroup,
+    },
+    include: {
+      model: models.Order,
+      required: true,
+      include: [models.Subscription],
+    },
+  });
+
+  if (transactions.length > 0) {
+    const order = paymentIntentTransaction.Order;
+
+    // closedReasons: approved, refunded, refunded_as_fraud, disputed, redacted
+    if (closedReason === 'refunded_as_fraud' || closedReason === 'refunded') {
+      if (order.status === OrderStatuses.IN_REVIEW) {
+        if (order.SubscriptionId) {
+          await order.update({ status: OrderStatuses.CANCELLED });
+        } else {
+          await order.update({ status: OrderStatuses.REFUNDED });
+        }
+      }
+
+      // Create refund transaction for the fraudulent charge
+      const transactionGroup = uuid();
+      await createRefundTransaction(
+        paymentIntentTransaction,
+        0,
+        {
+          ...paymentIntentTransaction.data,
+          review: event,
+          refundTransactionId: paymentIntentTransaction.id,
+        },
+        null,
+        transactionGroup,
+      );
```

Comment:
As pointed out in the demo, we need to make sure what exactly is being refunded and who's paying for the payment processor fees.